app/scripts/modules/core/src/managed/config/ManagementWarning.tsx

```diff
@@ -1,15 +1,18 @@
 import React from 'react';
 
-import { showModal } from 'core/presentation';
+import { useApplicationContextSafe, useShowModal } from 'core/presentation';
 
 import { ResumeManagementModal } from './Configuration';
 import { useFetchApplicationManagementStatusQuery, useToggleManagementMutation } from '../graphql/graphql-sdk';
 import { MODAL_MAX_WIDTH } from '../utils/defaults';
+
 import './ManagementWarning.less';
 
-export const ManagementWarning = ({ appName }: { appName: string }) => {
+export const ManagementWarning = () => {
+  const appName = useApplicationContextSafe().name;
   const { data, refetch } = useFetchApplicationManagementStatusQuery({ variables: { appName } });
   const [toggleManagement] = useToggleManagementMutation();
+  const showModal = useShowModal();
```

Comment:
It feels like the hook is adding a lot of code to do something simple. IMO it is more readable to fetch the application separately if it's needed, then pass the app context as a prop when invoking `showModal`. With the suggested approach (below), there is a consistent way to use `showModal` with or without the app prop, and the component will delegate which state is rendered based on the props. 

```suggestion
  const appContext = useApplicationContextSafe();
  const { data, refetch } = useFetchApplicationManagementStatusQuery({ variables: { appName: app.name } });
  const [toggleManagement] = useToggleManagementMutation();
  // later...
  showModal(ResumeManagementModal, props, { maxWidth: MODAL_MAX_WIDTH }, appContext);
```

Reply:
That was my original implementation but I thought that it could be useful to add this abstraction so the consumer of the modal won't have deal with that. Does make it sense?

Reply:
Can we call it `useShowModalWithAppContext`?